gwu-libraries/scholarspace-hyrax
# Initial tests

## Part 3/3 of initial RSpec setup

This PR accomplishes two things:
- Removing unused spec files. These were spec file templates that were generated during the initial creation of work-types, but were never filled in with test criteria. We could add them back later, but not serving a purpose here and preventing setup of CI/CD. 
- Adds tests for:
  - Dashboard content as admin vs. content admin
  - Depositing a PDF through dashboard workflow
  - Homepage contents and header/footer links. 
  - Sign-in/sign-out functionality. 
  - Sorting of catalog page by various options. 

We will need to continue adding to the tests, but these tests cover some of the major functionality of the app to start with.

## To test:
- Start docker containers. 
- Start interactive terminal in the app-server container, e.g. `docker exec -it scholarspace-hyrax_app-server_1 /bin/sh`
- Change to scholarspace user with `su scholarspace`
- Create dev/test databases with `bundle exec rails db:create`
- Migrate dev/test databases with `bundle exec rails db:migrate`
- Run `bundle exec rspec`

You should see 40 passing tests, and a coverage report ~56%. 

![image](https://github.com/gwu-libraries/scholarspace-hyrax/assets/17027357/cc66d357-a014-4d6d-8375-1114fc47f9dc)
